go/arrow/array/numeric.gen.go

```diff
@@ -80,6 +82,27 @@ func (a *Int64) setData(data *Data) {
 	}
 }
 
+func (a *Int64) getOneForMarshal(i int) interface{} {
+	if a.IsNull(i) {
+		return nil
+	}
+
+	return float64(a.values[i]) // prevent uint8 from being seen as binary data
```

Comment:
won't this truncate int64 values that require > 52 bits of storage?

Reply:
Fair, I'll update the template to only add that to the uint8 type

Reply:
fixed